kroxylicious-filters/kroxylicious-record-validation/src/main/java/io/kroxylicious/proxy/filter/validation/ProduceRequestValidatorBuilder.java

```diff
@@ -61,11 +67,41 @@ private static BytebufValidator toValidator(BytebufValidation valueRule) {
         valueRule.getSyntacticallyCorrectJsonConfig().ifPresent(config -> validators.add(BytebufValidators.jsonSyntaxValidator(config.isValidateObjectKeysUnique())));
         valueRule.getSchemaValidationConfig().ifPresent(
                 config -> validators.add(BytebufValidators.jsonSchemaValidator(
-                        Map.of(SchemaResolverConfig.REGISTRY_URL, config.apicurioRegistryUrl().toString()),
+                        buildSchemaResolverConfig(config),
                         config.apicurioContentId(),
                         config.wireFormatVersion())));
 
         return BytebufValidators.chainOf(validators);
     }
 
+    private static Map<String, Object> buildSchemaResolverConfig(SchemaValidationConfig config) {
+        Map<String, Object> resolverConfig = new HashMap<>();
+        resolverConfig.put(SchemaResolverConfig.REGISTRY_URL, config.apicurioRegistryUrl().toString());
+
+        Tls tls = config.tls();
+        if (tls != null) {
+            addTlsConfig(resolverConfig, tls);
+        }
+
+        return resolverConfig;
+    }
+
+    private static void addTlsConfig(Map<String, Object> resolverConfig, Tls tls) {
+        TrustProvider trustProvider = tls.trust();
+        if (trustProvider instanceof TrustStore trustStore) {
+            resolverConfig.put(SchemaResolverConfig.TLS_TRUSTSTORE_LOCATION, trustStore.storeFile());
+            if (trustStore.storePasswordProvider() != null) {
+                resolverConfig.put(SchemaResolverConfig.TLS_TRUSTSTORE_PASSWORD, trustStore.storePasswordProvider().getProvidedPassword());
+            }
+            if (trustStore.storeType() != null) {
+                resolverConfig.put(SchemaResolverConfig.TLS_TRUSTSTORE_TYPE, trustStore.storeType());
+            }
+        }
+        else if (trustProvider instanceof InsecureTls insecureTls && insecureTls.insecure()) {
+            resolverConfig.put(SchemaResolverConfig.TLS_TRUST_ALL, true);
+            resolverConfig.put(SchemaResolverConfig.TLS_VERIFY_HOST, false);
+        }
```

Comment:
better to make PlatformTrustProvider explicit in the if/then/else ladder and fail with an exception if you see a type you don't recognise.